framework/logging.cpp

```diff
@@ -1761,14 +1761,18 @@ void KeyValuePairLogger::print_child_stderr()
 void TapFormatLogger::print(int tc)
 {
     // build the ok / not ok line
-    const char *qual = quality_string(test);
-    const char *extra = nullptr;
+    std::string extra;
+    if (const char *qual = quality_string(test))
+        extra = qual;
+
+    const char *okstring = "not ok";
     switch (childExitStatus.result) {
     case TestSkipped:
     case TestPassed:
+        okstring = "ok";
         // recheck, as status.result does not take failing threads into account
         if (testResult == TestSkipped) {
-            extra = "SKIP";
+            extra += "SKIP";
```

Comment:
This could be set in `TestSkipped` and then we could fall through to `TestPassed`.

Reply:
Yes, but this was a minimal change. What you're suggesting is coming in a new change.